### ISSUE TITLE
why `stopCapture` waits for 1 second

### ISSUE THREAD

**Issue by tigercosmos** [seladb/PcapPlusPlus#1246]
why do we wait 1 second to stop the thread?
if there is any potential race condition, probably 100ms is enough? or could be even shorter.
However, I feel it's an anti-pattern to use `sleep` to ensure the behavior is correct.

```cpp
void PcapLiveDevice::stopCapture()
{
	// in blocking mode stop capture isn't relevant
	if (m_cbOnPacketArrivesBlockingMode != nullptr)
		return;

	m_StopThread = true;
	if (m_CaptureThreadStarted)
	{
		pcap_breakloop(m_PcapDescriptor);
		PCPP_LOG_DEBUG("Stopping capture thread, waiting for it to join...");
		m_CaptureThread.join();
		m_CaptureThreadStarted = false;
		PCPP_LOG_DEBUG("Capture thread stopped for device '" << m_Name << "'");
	}
	PCPP_LOG_DEBUG("Capture thread stopped for device '" << m_Name << "'");
	if (m_StatsThreadStarted)
	{
		PCPP_LOG_DEBUG("Stopping stats thread, waiting for it to join...");
		m_StatsThread.join();
		m_StatsThreadStarted = false;
		PCPP_LOG_DEBUG("Stats thread stopped for device '" << m_Name << "'");
	}

	multiPlatformSleep(1);
	m_StopThread = false;
}
```

cc @clementperon  @seladb

**Comment by seladb** [comment]
@tigercosmos this goes back all the way to the first commit in PcapPlusPlus: https://github.com/seladb/PcapPlusPlus/commit/488754ea464364e526630048ebbb8afc28ae4a05#diff-6dc93f9ea25d0bcd5957051301b157d18c254960c90fdef71e8a819ffb9cb273R281

It was 9 years ago so I don't remember why I added it, but probably to avoid a race condition when joining threads 🤷

**Comment by egecetin** [comment]
Fixed with #1247
